File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FValueAnimator.java
package io.github.gofaith.faithdroid.FViews;

import android.animation.ValueAnimator;
import android.util.Log;

import org.json.JSONArray;
import org.json.JSONTokener;

import java.util.ArrayList;
import java.util.List;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FValueAnimator extends FView implements AttrGettable, AttrSettable {
    public ValueAnimator v;

    public FValueAnimator(UIController uiController) {
        parentController=uiController;
        v=new ValueAnimator();
    }
    @Override
    public String getAttr(String attr) {
        return null;
    }

    @Override
    public void setAttr(String attr, final String value) {
        if (attr == null || value == null) {
            return;
        }
        Log.d(TAG, "setAttr: "+attr+","+value);
        switch (attr) {
            case "OfFloat":
                try {
                    JSONArray array = (JSONArray) (new JSONTokener(value).nextValue());
                    float[] vs = new float[array.length()];
                    for (int i=0;i<array.length();i++){
                        vs[i] = (float) array.getDouble(i);
                    }
                    v.setFloatValues(vs);
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "OfInt":
                try {
                    JSONArray array = (JSONArray) (new JSONTokener(value).nextValue());
                    int[] vs = new int[array.length()];
                    for (int i=0;i<array.length();i++){
                        vs[i] =  array.getInt(i);
                    }
                    v.setIntValues(vs);
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "Duration":
                try {
                    long d = Long.parseLong(value);
                    v.setDuration(d);
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "ValueChangedListener":
                v.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
                    @Override
                    public void onAnimationUpdate(ValueAnimator animation) {
                        Faithdroid.triggerEventHandler(value, animation.getAnimatedValue().toString());
                    }
                });
                break;
            case "Start":
                v.start();
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FValueAnimator.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.animation.ValueAnimator
import android.content.ContentValues.TAG
import android.util.Log
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.UIController
import org.json.JSONArray
import org.json.JSONTokener


class FValueAnimator(uiController: UIController) : FView(), AttrGettable, AttrSettable {
    var v: ValueAnimator

    init {
        parentController = uiController
        v = ValueAnimator()
    }

    override fun getAttr(attr: String): String{
        return ""
    }

    override fun setAttr(attr: String, value: String) {
        if (attr == null || value == null) {
            return
        }
        Log.d(TAG, "setAttr: $attr,$value")
        when (attr) {
            "OfFloat" -> try {
                val array = JSONTokener(value).nextValue() as JSONArray
                val vs = FloatArray(array.length())
                for (i in 0 until array.length()) {
                    vs[i] = array.getDouble(i).toFloat()
                }
                v.setFloatValues(*vs)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "OfInt" -> try {
                val array = JSONTokener(value).nextValue() as JSONArray
                val vs = IntArray(array.length())
                for (i in 0 until array.length()) {
                    vs[i] = array.getInt(i)
                }
                v.setIntValues(*vs)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "Duration" -> try {
                val d = java.lang.Long.parseLong(value)
                v.duration = d
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "ValueChangedListener" -> v.addUpdateListener { animation ->
                Faithdroid.triggerEventHandler(
                    value,
                    animation.animatedValue.toString()
                )
            }
            "Start" -> v.start()
        }
    }
}
